specification/dashboard/resource-manager/Microsoft.Dashboard/preview/2022-10-01-preview/grafana.json

```diff
@@ -0,0 +1,1998 @@
+        }
+      }
+    },
+    "ManagedGrafanaListResponse": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedGrafana"
+          }
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      }
+    },
+    "ManagedGrafanaProperties": {
+      "type": "object",
+      "description": "Properties specific to the grafana resource.",
+      "properties": {
+        "provisioningState": {
+          "description": "Provisioning state of the resource.",
+          "$ref": "#/definitions/ProvisioningState"
+        },
+        "grafanaVersion": {
+          "description": "The Grafana software version.",
+          "type": "string",
+          "readOnly": true
+        },
+        "endpoint": {
+          "description": "The endpoint of the Grafana instance.",
+          "type": "string",
+          "readOnly": true
+        },
+        "publicNetworkAccess": {
+          "description": "Indicate the state for enable or disable traffic over the public interface.",
+          "$ref": "#/definitions/PublicNetworkAccess"
+        },
+        "zoneRedundancy": {
+          "description": "The zone redundancy setting of the Grafana instance.",
+          "$ref": "#/definitions/ZoneRedundancy"
+        },
+        "apiKey": {
+          "description": "The api key setting of the Grafana instance.",
+          "$ref": "#/definitions/ApiKey"
+        },
+        "deterministicOutboundIP": {
+          "description": "Whether a Grafana instance uses deterministic outbound IPs.",
+          "$ref": "#/definitions/DeterministicOutboundIP"
+        },
+        "outboundIPs": {
+          "description": "List of outbound IPs if deterministicOutboundIP is enabled.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "readOnly": true
+        },
+        "privateEndpointConnections": {
+          "description": "The private endpoint connections of the Grafana instance.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnection"
+          },
+          "readOnly": true
+        },
+        "autoGeneratedDomainNameLabelScope": {
+          "description": "Scope for dns deterministic name hash calculation.",
+          "$ref": "#/definitions/AutoGeneratedDomainNameLabelScope"
+        },
+        "grafanaIntegrations": {
+          "$ref": "#/definitions/GrafanaIntegrations"
+        },
+        "enterpriseConfigurations": {
+          "$ref": "#/definitions/EnterpriseConfigurations"
+        },
+        "grafanaConfigurations": {
+          "$ref": "#/definitions/GrafanaConfigurations"
+        },
+        "grafanaPlugins": {
+          "description": "Installed plugin list of the Grafana instance. Key is plugin id, value is plugin definition.",
+          "type": "object",
+          "additionalProperties": {
```

Comment:
Use of additionalProperties is not allowed in ARM, unless for very specific scenarios. Why can't this be used as an array of objects (each object has plugin id and plugin resource model)?

Reply:
[must fix]

Reply:
We have discussed this at https://github.com/Azure/azure-rest-api-specs-pr/pull/14416
The main reason is based on plugin number and user operations. We expect user may have dozens or even hundreds of plugins, and we want to allow user use PATCH to simplify the operation. 
For example, one resource has 100 plugins, say plugin1 ~ plugin100. If user wants to remove plugin1 and install plugin101. If we use array, for PUT, user needs to list all plugins from 2 ~ 101 in payload, and PATCH does not work. Despite that, it has potential concurrent issue. If there is another user wants to remove plugin2 and install plugin102. One operation will override the other.
While if we use dict, user can do PATCH with only 2 plugins in payload -- plugin1's value as null, plugin101's value is plugin model. We can easily understand user's intention. And because we give delta, we can support concurrent operations. 
This pattern has been used by User Assigned Managed Identity for long time.

Add User Assigned Managed Identity
![image](https://github.com/Azure/azure-rest-api-specs/assets/7960665/05b5285f-f475-4f6a-9c9c-36795d734c6f)
Remove User Assigned Managed Identity
![image](https://github.com/Azure/azure-rest-api-specs/assets/7960665/802118e7-05d0-4694-9a85-b9eccb1cb422)

Reply:
Will not fix